### PR TITLE
Fix Native CI

### PR DESCRIPTION
**Pull request by armanbilge**
Faffing about in the dark, but I think it's a memory issue. So first try is to reduce the number of modules in the cats build.

### CODE REVIEW

**Comment by satorg** [comment]
I wonder if we also could extract cats-free into a separate repository (similar to cats-mtl).. To my perception, it looks quite different from all other modules and perhaps could be extracted out to reduce the overall weight of the cats repo.

**Comment by armanbilge** [comment]
It's a possibility, IDK. One of my first PRs to Cats was bringing in the `algebra` repository 😅  I'm not entirely sure what the criteria for inclusion/exclusion is (besides stability), what did you have in mind.

There's a maintenance overhead for every independent repository, although sbt-typelevel has helped lower that. On the other hand, there is a maintenance overhead for large monorepos, which Native has increased ...

File: build.sbt [hunk 1]
```diff
@@ -67,7 +67,10 @@ lazy val commonJsSettings = Seq(
   tlVersionIntroduced ++= List("2.12", "2.13").map(_ -> "2.1.0").toMap
 )
 
+lazy val NativeLink = Tags.Tag("native-link")
+Global / concurrentRestrictions += Tags.limit(NativeLink, 2)
 lazy val commonNativeSettings = Seq(
+  Test / nativeLink := (Test / nativeLink).tag(NativeLink).value,
```
Comment (armanbilge) [inline build.sbt:73]: This should be upstreamed :)
- https://github.com/scala-native/scala-native/issues/2853

**Comment by armanbilge** [comment]
Ok, I fixed it by adding a limit on number of Native linking tasks running concurrently in the build. It makes things go a lot slower I think :(

Not really sure what happened. Our runner was updated to ubuntu 22.04, and we either have less memory or maybe it ships with a new version of LLVM that is more memory hungry /shrug

**Comment by armanbilge** [comment]
Yeah, in https://github.com/typelevel/cats/commit/b036fb47196d1a788db09eb2bc262071fa3cb928 (which is one of the last CI jobs to run on ubuntu 20.04) the `nativeLink` step runs concurrently in about 10 minutes.

https://github.com/typelevel/cats/actions/runs/3615058132/jobs/6091912153

Now we are no longer running it concurrently and it takes 20 minutes. Trying one more thing ...

**Review by danicheg — APPROVED** [review]
Not quite following whether this is resolving the issue, but just take my 👍🏻

**Comment by armanbilge** [comment]
Thanks. Yep it resolved the issue (CI is working) but now it runs much more slowly :(

**Pull request merged**